src/main/java/com/aphyr/riemann/client/Promise.java

```diff
@@ -20,14 +21,37 @@ public static Object rehydrate(final Object value) throws IOException {
     }
   }
 
+  public static Object asResult(Object value) {
+    if (value instanceof Exception) {
+      return null;
+    }
+    else {
+      return value;
+    }
+  }
+
+  public static Exception asException(Object value){
+    if (value instanceof Exception) {
+      return (Exception) value;
+    }
+    else {
+      return null;
```

Comment:
Ditto here; need another channel for distinguishing between exceptions and promises. Is rehydrate() not appropriate?